src/spl/spline.h

```diff
@@ -236,8 +257,113 @@ class Spline {
     return total_length;
   }
 
+  std::vector<double> ComputeBezierDegreeElevationCoeffients(int dimension) const {
+    std::vector<double> alpha;
+    for (int i = 0; i < GetDegree(dimension).get() + 2; ++i) {
+      alpha.push_back(static_cast<double>(i) / (GetDegree(dimension).get() + 1));
+    }
+    return alpha;
+  }
+
+  std::vector<int> ProduceBezierSegments(int dimension) {
+    size_t position = GetKnotVector(dimension)->GetNumberOfKnots() - GetDegree(dimension).get() - 2;
+    std::vector<int> diff(GetKnotVector(dimension)->GetNumberOfDifferentKnots() - 2, GetDegree(dimension).get());
+    for (size_t i = GetKnotVector(dimension)->GetNumberOfDifferentKnots() - 2; i > 0; --i, --position) {
```

Comment:
No, there is no specific reason, I will change it to forward itertion. But I think replacing `i` by `position` will make it more complicated because `i` stands for the current knot span and `position` for the current knot. I think it's best to only change the names.